app/src/main/java/org/oppia/app/player/state/StateViewModel.kt

```diff
@@ -32,7 +33,14 @@ class StateViewModel @Inject constructor() : ObservableViewModel() {
 
   // TODO(#164): Add a hasPendingAnswer() that binds to the enabled state of the Submit button.
   fun getPendingAnswer(): UserAnswer {
-    return getPendingAnswerHandler(itemList)?.getPendingAnswer() ?: UserAnswer.getDefaultInstance()
+    if (getPendingAnswerHandler(itemList) is FractionInteractionViewModel && (getPendingAnswerHandler(itemList) as FractionInteractionViewModel).viewType.name.equals("FRACTION_INPUT_INTERACTION")) {
```

Comment:
We should not be special casing the fraction view model here, otherwise it breaks the generic design pattern being used here. We should find a generic way to accomplish what's being done here, instead.

Reply:
Done. PTAL